cpp/modmesh/buffer/BufferExpander.hpp

```diff
@@ -43,8 +44,8 @@ namespace modmesh
  * Untyped and growing memory buffer for contiguous data storage.  The internal
  * expandable memory buffer cannot be used externally.
  */
-class BufferExpander
-    : public std::enable_shared_from_this<BufferExpander>
+class BufferExpander : public BufferBase<BufferExpander>
```

Comment:
`std::enable_shared_from_this` should be the first parent.

Use a standalone line for `public BufferBase<BufferExpander>` to make the block like:

```cpp
class BufferExpander
    : public BufferBase<BufferExpander>
    , public std::enable_shared_from_this<BufferExpander>
```

Reply:
I think `std::enable_shared_from_this` was the first parent already?
fixed the foramt.